contracts/test-integration/BlsProviderContractInteraction.test.ts

```diff
@@ -161,6 +161,9 @@ describe("Provider tests", function () {
         .transfer(recipient, amountToTransfer);
       await tx.wait();
 
+      // wait 1 second to ensure listener count updates
+      await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
When running the whole test suite locally, the `expect(blsProvider.listenerCount(filter)).to.equal(0);` assertion would periodically fail as the event listener was still subscribed. Seems a bit hacky for testing a function that is supposed to be synchronous, but the timer appears to fix that issue.